Mathlib/Topology/MetricSpace/Holder.lean

```diff
@@ -211,6 +292,12 @@ lemma mono {C' : ℝ≥0} (hf : HolderWith C r f) (h : C ≤ C') :
     HolderWith C' r f :=
   fun x₁ x₂ ↦ (hf x₁ x₂).trans (by gcongr)
 
+/-- If a function is `r`-Hölder over a bounded space, then it is also `t`-Hölder when `t ≤ r`. -/
+lemma of_le {C D t : ℝ≥0}
+    (hs : ∀ x y : X, edist x y ≤ D) (hf : HolderWith C r f) (htr : t ≤ r) :
+    HolderWith (C * D ^ (r - t : ℝ)) t f :=
+  holderOnWith_univ.1 ((holderOnWith_univ.2 hf).of_le (fun x _ y _ ↦ hs x y) htr)
```

Comment:
Sorry to be a bit annoying, but would you mind specializing the other results to `HolderWith` as well?